src/librustdoc/html/format.rs

```diff
@@ -244,8 +244,8 @@ impl<'a> fmt::Display for WhereClause<'a> {
                 clause.push_str("<br>");
             }
 
-            match pred {
-                &clean::WherePredicate::BoundPredicate { ref ty, ref bounds } => {
+            match *pred {
+                clean::WherePredicate::BoundPredicate { ref ty, ref bounds } => {
```

Comment:
Unneeded deref and unneeded `ref` keywords. Same for the other arms of this `match`.